### ISSUE TITLE
Embedded framework with Resources dependency

### ISSUE THREAD

**Issue by Zyphrax** [Carthage/Carthage#1959]
A few days ago I took on the job of replacing CocoaPods with Carthage in our app. A few tweaks here and there but I have most of our dependencies working. However I'm really puzzled to get one of them to work: [SumUp SDK](https://github.com/sumup/sumup-ios-sdk).

The SumUp SDK is a precompiled closed-source framework, with two resource bundles:
```
SumupSDK.embeddedframework
  SumupSDK.framework
  Resources
     SMPSharedResources.bundle
     YTLibResources.bundle
```

To integrate SumUp with CocoaPods I forked the project, added a podspec and a modulemap, and a few hours of trial and error later, SumUp was working well in my Swift project.

For Carthage I was trying to wrap the SumupSDK.framework + resource bundles in a Cocoa Touch Framework library, but judging by the file size of the build result the framework isn't embedded into my wrapper framework.

I've already spend hours on this, I hope that you guys can help me towards a solution.

**Comment by mdiep** [comment]
You shouldn't directly embed the SDK into your wrapper framework. You should _link_ against it, but only copy the SDK into your app target. https://github.com/Carthage/Carthage#nested-dependencies

**Comment by Zyphrax** [comment]
@mdiep thank you. I've looked at the nested dependencies section, but I'm not really sure how it applies. In that case I would use a Cartfile to link the SumupSDK to my wrapper framework?

The SumUp SDK looks like a static framework:
```
SumupSDK: Mach-O universal binary with 4 architectures: [arm_v7: current ar archive] [i386] [x86_64] [arm64]
SumupSDK (for architecture armv7):	current ar archive
SumupSDK (for architecture i386):	current ar archive
SumupSDK (for architecture x86_64):	current ar archive
SumupSDK (for architecture arm64):	current ar archive
```

Should I wrap that in a dynamic framework (Cocoa Touch Framework)?

I don't like the separate resource bundles. It would be great if I could combine it into a framework (call it SumUp) and simply add that to my project. Just like the other frameworks.

I am really puzzled on how to integrate this.

**Comment by mdiep** [comment]
Ah, if it's a static framework, then you should be ably to link against it in your wrapper and copy any required resources into your framework bundle.

You should contact the SDK vendor. It sounds like they're using bundles because they're using a static framework—since they don't have a section for bundled resources. If they provided a dynamic framework, that would likely solve your issues.

**Comment by Zyphrax** [comment]
I've contacted the vendor, but unfortunately it will probably take months before they'll fix it. Some requests like CocoaPods support are almost 2 years old, although it took me a few hours to create CocoaPods support myself.

Can I convert the static framework into a dynamic framework?
They don't release updates often.

Or would the wrapper work? Would that basically turn it into a dynamic framework?
In that case I would add a Cartfile to my dynamic wrapper framework and use a binary reference to their zip file? Would it properly recognize the framework? And what would Carthage do with the resource bundles?

**Comment by mdiep** [comment]
I think you could create a dynamic framework, link it against their static framework, and copy the `.bundle`s into the resources section of your dynamic framework. (Or you might need to copy them into your applications resources section.)

I think that should work, although I've never tried it. Unfortunately I can't provide more support than that. Sorry.

**Comment by Zyphrax** [comment]
After countless attempts I've managed to create a wrapper around the SumUp SDK 🎉 
Now I have to fix the resources, but I think I can copy most of CocoaPods build script for that.

**Comment by Zyphrax** [comment]
Instead of using a build script for the resource bundles, I've just dragged the two bundles from the framework folder into Xcode. Everything is working!

**Comment by HassanSE** [comment]
I'm having a similar issue. I've compressed a folder containing `.framework` files along with `.bundle` resource. Compiled a `.json` file:

`{
    "5.0.0": "{s3-bucket-url/path/framework.zip}"
}`

Now I hosted this `.json` on my S3 bucket. Added this line to my `Cartfile`:

`binary "{s3-bucket-url/path/framework.json}"`

On `carthage update --platform iOS`, I'm getting all the `.frameworks` within the `.zip` file but not getting `.bundle`, however, this `.bundle` was placed in the folder where other `.framework` files were placed. I don't want to manually place `.bundle` file every time a build needs to roll out on CI server (even with shell script). What's the best practice here? Thanks.
